### DISCUSSION TITLE
[Feature Request] Toolbar -> Use Lucide Icons as Items Label instead of Text

### DISCUSSION THREAD

**Discussion by FelipeRearden** [chrisgurney/obsidian-note-toolbar#10] [Ideas]
Hello @chrisgurney !!!!

This FR is about the UI of the Note Toolbar. To keep the UI a little bit more aligned with the Obsidian Native UI, I would like to have an option to use the native Obsidian Icons Library Lucide instead of the Item text. 

www.lucide.dev

📸 To share my idea, I created this mock-up
![IMG_1153](https://github.com/chrisgurney/obsidian-note-toolbar/assets/85180696/fd407a35-0b6e-4f1d-90d8-0ca2c1f48bac)

ℹ️ A new text field to let the user write the name of the lucide icon that will represent the Item. Optional text field;

ℹ️ An checkbox that hides the Item name from the Note Toolbar when checked. This way the user could have three options:
1- Item Name
2- Item Name + Icon 
3- Icon

<hr>

I hope you like this idea 🙏 🙏 🙏

Let me know what you think !!!!

**Comment by chrisgurney** [comment]
This idea makes total sense, as well! It will take me a bit more time to implement this one, but I'm definitely adding it to my list. Thanks!

Here's my first crack at what I'll have to do here:

- [x] look at Shell Commands for getting the list of Lucide icons → hard-coded: https://github.com/Taitava/obsidian-shellcommands/blob/main/src/Icons.ts
  - [x] possible to instead generate a list somehow?
- [x] create IconSuggester class
- [x] icon only = just a set icon with no label text?
- [x] add item setting for icon
- [x] try displaying lucide icon in editor (try same way rendered in editor — or is that just an SVG?)
  - [ ] can use shell commands’ approach here: https://github.com/Taitava/obsidian-shellcommands/blob/main/src/Icons.ts … but may have to look for alternative to using innerHTML
- [x] add Setting UI
  - [x] empty state = box? dotted box CSS?
- [x] migration to add item setting

**Reply by FelipeRearden** [reply]
> This idea makes total sense, as well! It will take me a bit more time to implement this one, but I'm definitely adding it to my list. Thanks!

Great ! Thank you very much for the feedback!!!

> Here's my first crack at what I'll have to do here:

Sharing some information that might be useful 🙏

> look at Shell Commands for getting the list of Lucide icons
The plugin Obsidian Commander Pluign has a Lucide Icons modal that is used to define an icon to a command 
Might be exactly what you a looking for 

https://github.com/phibr0/obsidian-commander

> try displaying lucide icon in editor (try same way rendered in editor — or is that just an SVG?)

The plugin Badges lets you render Lucide icons in the Editor uisng inline code syntax. 
Maybe could save you some time 🙏

https://github.com/gapmiss/badges

<hr>

I hope it helps 🙏 🙏 🙏

**Reply by chrisgurney** [reply]
Looks useful, thank you!

**Comment by chrisgurney** [comment]
Now available in the beta! [Check it out.](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.4.0)

**Reply by FelipeRearden** [reply]
Thank you very much for the new beta release @chrisgurney  !!!!

✅ Version 1.3.8 (1.4.1 Beta) installation went perfect on Obsidian Mobile.

📋 I am gonna test the icons settings and the new and CSS to create my custom note toolbar
📋 I am gonna test the `Command` item option (I am very excited about this too)

I get back to you in a few hours and forgive for my late reply today ☀️ 
It’s gonna be amazing ! I am dreaming about this icons since the first time I use your plugin 🙏

**Reply by FelipeRearden** [reply]
Working perfect on version `1.4.1` on Obsidian Mobile. 
✅ The UI is amazing and very easy to interact with
✅ `Command` feature is working perfect as well
✅ The CSS for the icons is easy to customize 🎉

Thank you very much for this amazing feature !!!!!!

I get back to this issue after the merge to test again :)

**Reply by chrisgurney** [reply]
Icon selection is even better now on mobile. 😄 
https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.4.3
